packages/rum-core/test/performance-monitoring/performance-monitoring.spec.js

```diff
@@ -763,4 +764,51 @@ describe('PerformanceMonitoring', function() {
     performanceMonitoring.processAPICalls('invoke', task3)
     expect(tr.ended).toBe(true)
   })
+
+  it('should send span context destination details to apm-server', done => {
+    const transactionService = serviceFactory.getService('TransactionService')
+
+    const tr = transactionService.startTransaction('with-context', 'custom')
+    const data = {
+      url: 'http://localhost:3000/b/c',
+      method: 'GET',
+      target: {
+        status: 200
+      }
+    }
+    const xhrSpan = tr.startSpan(`GET ${data.url}`, 'external')
+    xhrSpan.end(null, data)
+
+    const resourceUrl = 'http://example.com'
+    const rtData = {
+      url: resourceUrl,
+      entry: resourceEntries.filter(({ name }) => name === resourceUrl)[0]
+    }
+    const rtSpan = tr.startSpan(rtData.url, 'resource')
+    rtSpan.end(null, rtData)
+
+    configService.events.observe(TRANSACTION_END, () => {
+      expect(tr.spans.length).toBe(2)
+      const [span1, span2] = tr.spans
+      expect(span1.context.destination).toBeDefined()
+      expect(span2.context.destination).toBeDefined()
+
+      const payload = performanceMonitoring.convertTransactionsToServerModel([
+        tr
+      ])
+      expect(payload[0].spans[0].context.destination).toBeDefined()
+      expect(payload[0].spans[1].context.destination).toBeDefined()
+
+      apmServer
+        .sendTransactions(payload)
+        .catch(reason => {
+          fail(
```

Comment:
Please use `done.fail()` instead, as discussed in previous PRs, this won't end test properly.

Reply:
Wow didn't know something like that exists, for sure 👍